src/store/index.js

```diff
@@ -34,25 +38,76 @@ export default new Vuex.Store({
       state.collections = await state.connection.db.listCollections().toArray()
     },
     async find({ state }, query = {}) {
-      state.totalNumberOfResults = null
+      if (state.collectionRecords[state.collectionName]) {
+        const query_serialization = JSON.stringify(query)
+
+        if (state.collectionRecords[state.collectionName].query === query_serialization) {
+          state.activeRecords = state.collectionRecords[state.collectionName]
+          state.totalNumberOfResults = state.activeRecords.result.length
+          return
+        }
+      }
+
+      state.totalNumberOfResult = null
       const collection = state.connection.db.collection(state.collectionName)
       const cursor = await collection.find(query)
+
       cursor.count((err, result) => {
         if (err) return
-
         state.totalNumberOfResults = result
       })
-      state.records = await cursor.toArray()
+
+      const result = await cursor.toArray()
+
+      state.activeRecords = {
+        query: JSON.stringify(query),
+        result,
+        isEmpty: !result.length
+      }
+
+      state.collectionRecords[state.collectionName] = state.activeRecords
     },
     async setDatabase({ state, dispatch }, databaseName) {
       state.connection = state.connection.useDb(databaseName)
       state.databaseName = databaseName
       state.collectionName = ''
-      state.records = []
+      state.collectionRecords = {}
+      state.openedCollections = []
```

Comment:
this should simply be `Object.keys(state.collectionRecords)`, no?

Reply:
I prefer to keep it different variable because I'm using for opened tabs and this way helps me easy to read, update  :)